### DISCUSSION TITLE
Dynamic images import with Quasar CLI (with vite)

### DISCUSSION THREAD

**Discussion by phgrund** [quasarframework/quasar#14154] [General - Components / Directives / etc] [answered]
I'm currently migrating my project from webpack to vite, followed the [official guide](https://quasar.dev/quasar-cli-vite/convert-to-quasar-cli-with-vite) and everything seems to work just fine, except images with dynamic URLs, here are the two cases I have going on:

1. With require()
```
computed(() => {
  return [
      { icon: require('src/assets/emojis/stopwatch.png') },
      { icon: require('src/assets/emojis/stop-sign.png') },
      { icon: require('src/assets/emojis/receipt.png') }
    ]
})
```
I tried to change to import(), but doesn't seem to work
When changing to import and replacing with `~assets/...` the build fails

2. With string
```
computed(() => {
  return $q.dark.isActive || props.dark
    ? props.mini
      ? 'logos/docstage-mini-dark.svg'
      : 'logos/docstage-dark.svg'
    : props.mini
      ? 'logos/docstage-mini-light.svg'
      : 'logos/docstage-light.svg'
})
```
It doesn't throw any errors, but no image is displayed (I tried moving to assets folder but no success)

Any solutions?

**Comment by pdanpdan** [comment, marked as answer]
https://vitejs.dev/guide/features.html
Check glob import section

**Reply by phgrund** [reply]
Thanks! I ended up importing every image one by one as shown in the docs, but in a near future I'll replace with a dynamic solution
